File: sendlogs-ruby/sendlogs.rb
##########################################################
#    Docker: https://hub.docker.com/_/ruby/
#    docker run -it --rm --name my-running-script -v "$PWD":/usr/src/myapp -w /usr/src/myapp ruby:2.5 ruby your-daemon-or-script.rb
#

require 'net/http'
require 'uri'
require 'json'
require 'securerandom'
require 'ostruct'
require 'logger'

log_file = File.new("sendlogs.log", "w")
$logger = Logger.new log_file
$logger.level = Logger::DEBUG
$logger.formatter = proc do |severity, datetime, progname, msg|
   "NICE: #{msg}\n"
end

def post_request(logs)  
  
  #uri = URI.parse("http://localhost:8080/api/v1/logs")
  uri = URI.parse("https://api.coralogix.com/api/v1/logs")
  https = Net::HTTP.new(uri.host, uri.port)
  https.use_ssl = true
  header = {'Content-Type': 'application/json'}
  request = Net::HTTP::Post.new(uri.request_uri, header)  
  request.body = logs
  begin 
    response = https.request(request)   
	$logger.info('POST request sent')
	rescue Exception => e   
		$logger.info('POST request - error')
		exit
	end 
	$logger.info("Result " + response.code  + ", " + response.body + ", " + response.message + ", " + response.class.name)
  end


def generate_log_entries  
  log_entity_object = {
  'timestamp' => Time.now.to_f*1000,
  'severity' => 1 + rand(4),
  'text' => 'Encountered an error while ' + rand(36**12).to_s(36),
  'category' => [*('A'..'Z')].sample(3).join,
  'className' => 'ClassName-' + [*('a'..'z')].sample(10).join,
  'methodName' => 'MethodName-' + [*('a'..'z')].sample(10).join,
  'threadId' => [*('a'..'z')].sample(1).join + '-' + (100 + rand(900)).to_s
  }
  log_entity_object
end

def append_log_entries
  log_entity_list = Array.new 
  1000.times do
    log_entity_list.push(generate_log_entries)
  end  
  log_entity_list
end

def send_logs
  log_entries_json = '{
           "privateKey": "24cdd646-9f5d-3ff4-c567-d78496d27669",
           "applicationName": "RUBY sending logs",
           "subsystemName": "RUBY subsystem name",
           "computerName": "RUBY computer name",
           "logEntries": [
             {
               "timestamp": 1457827957703.342, 
               "severity": 4,
               "text": "Encountered an error while registering the user john123",
               "category": "DAL",
               "className": "UserManager",
               "methodName": "RegisterUser",
               "threadId": "a-352"
             }
           ]
        }'
        
  json_object = JSON.parse(log_entries_json)
  json_object["logEntries"] = append_log_entries
  3.times do  
	post_request(json_object.to_json)
  end
  
end



send_logs

File: sendlogs-python/sendlogs.py
import json
import time
import string
import requests
import logging
from random import *

logging.basicConfig(filename='sendlogs.log', level=logging.DEBUG)

json_template = json.loads(""" {
           "privateKey": "24cdd646-9f5d-3ff4-c567-d78496d27669",
           "applicationName": "PYTHON send logs",
           "subsystemName": "PYTHON subsystem name",
           "computerName": "PYTHON computer name",
           "logEntries": [
             {
               "timestamp": 1457827957703.342, 
               "severity": 4,
               "text": "Encountered an error while registering the user john123",
               "category": "DAL",
               "className": "UserManager",
               "methodName": "RegisterUser",
               "threadId": "a-352"
             }
           ]
        }""")

def generate():
    logEntry = {
    "timestamp": time.time() * 1000,
    "severity": randint(1, 6),
    "text" : 'Encountered an error while operation ' + ''.join(choice(string.ascii_letters + string.digits) for x in range(20)),
    "category": ''.join(choice(string.ascii_uppercase) for x in range(3)),
    "className": 'ClassName-' + ''.join(choice(string.ascii_letters) for x in range(10)),
    "methodName": 'MethodName-' + ''.join(choice(string.ascii_letters) for x in range(10)),
    "threadId" : 'x-' + str(randint(100, 999))
    }
    return logEntry

list = []
for x in range(0, 1001):
    list.append(generate())

json_template["logEntries"] = list
json_obj = json.dumps(json_template)
logging.info("JSON obj extended")
header = {'Content-type': 'application/json'}

for x in range(0, 3):
    r = None
    try:
        #r = requests.post("http://localhost:8080/api/v1/logs", data=json_obj, headers=header)
        r = requests.post("https://api.coralogix.com/api/v1/logs", data=json_obj, headers=header, verify=False)
        logging.info('POST request sent' + str(r.status_code) + ', ' +  str(r.reason) + ', ' + str(r.text))
    except requests.exceptions.RequestException as e:
        logging.info(e)

